src/Analyzers/CSharp/Analyzers/UseConditionalExpression/CSharpUseConditionalExpressionForReturnDiagnosticAnalyzer.cs

```diff
@@ -21,5 +22,43 @@ public CSharpUseConditionalExpressionForReturnDiagnosticAnalyzer()
 
         protected override ISyntaxFacts GetSyntaxFacts()
             => CSharpSyntaxFacts.Instance;
+
+        protected override (bool matched, bool canSimplify) TryMatchPattern(IConditionalOperation ifOperation, ISymbol containingSymbol)
+        {
+            if (!UseConditionalExpressionForReturnHelpers.TryMatchPattern(
+                    GetSyntaxFacts(), ifOperation, containingSymbol, out var isRef, out var trueStatement, out var falseStatement, out var trueReturn, out var falseReturn))
+            {
+                return default;
+            }
+
+            // Return statements wrapped in an unsafe, checked or unchecked block are not supported
+            if (IsWrappedByCheckedOrUnsafe(trueStatement) ||
+                IsWrappedByCheckedOrUnsafe(falseStatement))
+            {
+                return default;
+            }
+
+            var canSimplify = UseConditionalExpressionHelpers.CanSimplify(
+                trueReturn?.ReturnedValue ?? trueStatement,
+                falseReturn?.ReturnedValue ?? falseStatement,
+                isRef,
+                out _);
+
+            return (matched: true, canSimplify);
```

Comment:
this looks to be repeating what's in teh base.  Can you instead just call into the base, and *if* it matches, you then do the additional checks?

alternatively, have the base method be selaed, but which calls into a new virtual method (that is passed the true/false statements) that you can then check?

Reply:
Done, did the latter.